gpytorch/priors/torch_priors.py

```diff
@@ -118,4 +118,12 @@ def cpu(self):
         return module
 
     def expand(self, batch_shape):
-        return MultivariateNormal.expand(self, batch_shape, _instance=self)
+        from IPython.core.debugger import set_trace
+
+        set_trace()
+        batch_shape = torch.Size(batch_shape)
+        cov_shape = batch_shape + self.event_shape
+        new_loc = self.loc.expand(batch_shape)
```

Comment:
Should this be `self.loc.expand(batch_shape + torch.Size([self.loc.size(-1)]))`?

Reply:
@gpleiss Ugh. For the code repo as it is it is correct -- I went rogue and made `Prior`'s expand behave like tensors rather than distributions. e.g., to turn `a = torch.randn(3)` in to `1 x 3` you do `a.expand([1, 3])`, not `a.expand([1])` (same for `LazyTensors`).

If we wanted priors to follow the expand semantics of torch Distribution, we'd need to do some hackery here, where I currently call expand on the shape of the underlying Parameter so that the samples always have the same shape as the Parameter: https://github.com/cornellius-gp/gpytorch/blob/707bd988b3b22f0689a81177cea8ee681ebbf7f9/gpytorch/module.py#L395

This might involve adding batch_shapes and event_shapes or something to the parameters themselves, unless you have an obvious clean solution...

Reply:
Nope - let's keep your rogue solution :)